File: python/3254.find-the-power-of-k-size-subarrays-i.py
#
# @lc app=leetcode id=3254 lang=python3
#
# [3254] Find the Power of K-Size Subarrays I
#

# @lc code=start
class Solution:
    def resultsArray(self, nums: List[int], k: int) -> List[int]:
        n = len(nums)
        results = []

        for i in range(n - k + 1):
            subarr = nums[i:i+k]

            if all(subarr[j] + 1 == subarr[j + 1] for j in range(k - 1)):
                results.append(max(subarr))
            else:
                results.append(-1)

        return results

# @lc code=end

File: rust/3254.find-the-power-of-k-size-subarrays-i.rs
/*
 * @lc app=leetcode id=3254 lang=rust
 *
 * [3254] Find the Power of K-Size Subarrays I
 */

// @lc code=start
impl Solution {
    pub fn results_array(nums: Vec<i32>, k: i32) -> Vec<i32> {
        nums.windows(k as usize)
            .map(|subarr| {
                if subarr.windows(2).all(|pair| pair[1] == pair[0] + 1) {
                    *subarr.iter().max().unwrap()
                } else {
                    -1
                }
            })
            .collect()
    }
}

// @lc code=end

